src/print.js

```diff
@@ -1,42 +1,134 @@
+      </ul>
+      `,
+    );
+    localStorage.setItem('todoList', JSON.stringify(this.todoList));
+    // Add a click event listener to the remove button for the new todo item
+    const $removeButton = document.getElementById(`${item.description}`);
+    $removeButton.addEventListener('click', () => {
+      const indexToRemove = this.todoList.findIndex((e) => e.description === item.description);
+      if (indexToRemove > -1) {
+        this.remove(indexToRemove);
+        this.updateIndexes();
+      }
+    });
+    this.updateIndexes(); // Update the indexes after adding an item
+
+    // Add a click event listener to the edit button for the new todo item
+    // Edit button
+    const $editButton = document.getElementById(`edit-${item.description}`);
+    $editButton.addEventListener('click', () => {
+      const $inputField = document.getElementById('input'); // input div
+      $inputField.value = item.description;
+      const $addButton = document.getElementById('addBtn'); // add button
+
+      // Edit button click handler
+      $addButton.addEventListener('click', () => {
+        const newDescription = $inputField.value;
+        const indexToEdit = this.todoList.findIndex((e) => e.description === item.description);
+
+        if (newDescription !== '' && newDescription !== item.description) {
+          if (indexToEdit > -1) {
+            // Edit item
+            this.edit(indexToEdit, newDescription);
+            $editButton.id = `edit-${newDescription}`;
+            $editButton.textContent = newDescription;
+            this.updateIndexes();
+          }
+        } else {
+          // Delete item previous item
+          this.remove(indexToEdit);
+          $editButton.parentElement.removeChild($editButton);
+        }
+      });
+    });
+  };
+
+  // Define a remove method to remove a todo item from the DOM and the todoList array
+  remove = (indexToRemove) => {
+    const { parentElement } = document.getElementById(`${this.todoList[indexToRemove].description}`);
+    if (parentElement) {
+      parentElement.remove();
+    }
+
+    if (indexToRemove > -1) {
+      this.todoList.splice(indexToRemove, 1);
+      this.updateIndexes();
+      localStorage.setItem('todoList', JSON.stringify(this.todoList));
+      this.updateIndexes(); // Update the indexes after removing an item
+    }
+  };
+
+  edit = (indexToEdit, newDescription) => {
+    // Retrieve the todo item to edit from the todoList array
+    const todoToEdit = this.todoList[indexToEdit];
+
+    // If the todo item exists, update its description and update the DOM
+    if (todoToEdit) {
+      todoToEdit.description = newDescription;
+      const $todoDescription = document.querySelector(`.textTodo[data-index="${indexToEdit}"]`);
+      if ($todoDescription) {
+        $todoDescription.textContent = newDescription;
+        this.updateIndexes();
+      }
+      localStorage.setItem('todoList', JSON.stringify(this.todoList));
+    }
+  };
```

Comment:
- [ ] @emarkees Great work implementing the changes requested however the edit future still doesn't work, and there is no way to edit the todos, kindly ensure your edit functionality is working as previously requested.

Reply:
@danmainah The future is working from my end, I will upload a clip of it,